lib/ansible/modules/dnf.py

```diff
@@ -873,20 +859,8 @@ def _update_only(self, pkgs):
         return not_installed
 
     def _install_remote_rpms(self, filenames):
-        if int(dnf.__version__.split(".")[0]) >= 2:
-            pkgs = list(sorted(self.base.add_remote_rpms(list(filenames)), reverse=True))
-        else:
-            pkgs = []
-            try:
-                for filename in filenames:
-                    pkgs.append(self.base.add_remote_rpm(filename))
-            except IOError as e:
-                if to_text("Can not load RPM file") in to_text(e):
-                    self.module.fail_json(
-                        msg="Error occurred attempting remote rpm install of package: {0}. {1}".format(filename, to_native(e)),
-                        results=[],
-                        rc=1,
-                    )
+        pkgs = self.base.add_remote_rpms(filenames)
```

Comment:
Was `pkgs = list(sorted(self.base.add_remote_rpms(list(filenames)), reverse=True))` even necessary with dnf 2? I see it was added in https://github.com/ansible/ansible-modules-extras/pull/3325, but it doesn't seem like `add_remote_rpms` was mutating the input paths or returning the packages in the reverse order.

I don't see IOError handled in the outer scope, I think we might need to catch that still.

Reply:
> Was `pkgs = list(sorted(self.base.add_remote_rpms(list(filenames)), reverse=True))` even necessary with dnf 2? I see it was added in [ansible/ansible-modules-extras#3325](https://github.com/ansible/ansible-modules-extras/pull/3325), but it doesn't seem like `add_remote_rpms` was mutating the input paths or returning the packages in the reverse order.

Yeah I didn't see a reason why `filenames` were converted to a `list` (it already was), why the result of `sorted` was converted to a list (`sorted` already returns a `list`), why the sort was in the reverse order and why the packages were even sorted, so I just removed all of it :-) I looked at the dnf codebase and none of it is done when calling `add_remote_rpms` there.
 
> I don't see IOError handled in the outer scope, I think we might need to catch that still.

Good catch. I didn't care to check what exceptions `add_remote_rpms` raises since the original code didn't catch any. I reused the `except` from the loop and refactored the code a bit. Let me know what you think.

Thanks!

Reply:
Thanks for the confirmation, I spent a while squinting at a that :) The exception handling looks good to me!